sdks/common/fn-api/src/main/proto/beam_fn_api.proto

```diff
@@ -540,91 +536,82 @@ service BeamFnState {
   ) {}
 }
 
+message CacheToken {
+  // (Required) Represents the function spec and tag associated with this state
+  // key.
+  //
+  // By combining the function_spec_reference with the tag representing:
+  //   * the input, we refer to the iterable portion of a large GBK
+  //   * the side input, we refer to the side input
+  //   * the user state, we refer to user state
+  Target target = 1;
 
-// TODO: Resolve with the other State API.
-service SimpleBeamFnState {
-  // Gets the elements associated with the given key.
-  rpc Get(StateKey) returns (Elements.Data) {}
-  // Appends elements to a given state bag.
-  rpc Append(SimpleStateAppendRequest) returns (Empty) {}
-  // Clears a given state bag.
-  rpc Clear(StateKey) returns (Empty) {}
-}
-
-message Empty {
-}
-
-message SimpleStateAppendRequest {
-  StateKey state_key = 1;
-  repeated bytes data = 2;
+  // (Required) An opaque identifier.
+  bytes token = 2;
 }
 
 message StateKey {
-  // (Required) Represents the namespace for the state. If this state is for a
-  // DoFn, then this reference is expected to point to the DoFn. If this state
-  // is for a side input, then this is expected to reference the ViewFn.
-  string function_spec_reference = 1;
+  // (Required) Represents the function spec and tag associated with this state
+  // key.
+  //
+  // By combining the function_spec_reference with the tag representing:
+  //   * the input, we refer to fetching the iterable portion of a large GBK
+  //   * the side input, we refer to fetching the side input
+  //   * the user state, we refer to fetching user state
+  Target target = 1;
 
   // (Required) The bytes of the window which this state request is for encoded
-  // in the outer context.
+  // in the nested context.
   bytes window = 2;
 
-  // (Required) The user key for which the value was encoded in the outer
-  // context.
+  // (Required) The user key encoded in the nested context.
   bytes key = 3;
 }
 
-message StateKeyOrIterable {
-  // One of the two fields below are required to be set.
-  // If state key is set, then the State API should be invoked to fetch the
-  // values allowing one to restart the iterable. Otherwise the bytes for the
-  // entire iterable are represented and should be decoded using an iterable
-  // coder using the outer context.
-  StateKey state_key = 1;
-  repeated bytes iterable = 2;
+// A logical byte stream which can be continued using the state API.
+message ContinuableStream {
+  // An empty continuation token represents the end of the logical byte stream.
```

Comment:
It turns out that I meant to say that unsef = EOF, and not empty byte array = EOF